src/components/WalletSelector/multichain/tezos/TezosAuthenticateWallet.tsx

```diff
@@ -0,0 +1,157 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { BaseM, TitleS } from 'components/core/Text/Text';
+import { useAuthActions } from 'contexts/auth/AuthContext';
+import { INITIAL, PROMPT_SIGNATURE, PendingState } from 'types/Wallet';
+import Spacer from 'components/core/Spacer/Spacer';
+import {
+  isEarlyAccessError,
+  useTrackSignInAttempt,
+  useTrackSignInError,
+  useTrackSignInSuccess,
+} from 'contexts/analytics/authUtil';
+import { captureException } from '@sentry/nextjs';
+import useCreateNonce from 'components/WalletSelector/mutations/useCreateNonce';
+import useLoginOrRedirectToOnboarding from 'components/WalletSelector/mutations/useLoginOrRedirectToOnboarding';
+import { EthereumError } from '../EthereumError';
+import { normalizeError } from '../normalizeError';
+import { DAppClient, RequestSignPayloadInput, SigningType } from '@airgap/beacon-sdk';
+import { char2Bytes } from '@taquito/utils';
+
+type Props = {
+  reset: () => void;
+};
+
+export const TezosAuthenticateWallet = ({ reset }: Props) => {
+  const beaconClient = useMemo(() => {
+    return new DAppClient({ name: 'Gallery' });
```

Comment:
i believe this might initiate a new dapp client each time the component unmounts / remounts. i think it would be safest to simply define the client above the component:

```js
const beaconClient = new DAppClient({ name: 'Gallery' });

export const TezosAuthenticateWallet ...
```